### PR TITLE
[0.18] Backport #15911: Use wallet RBF default for walletcreatefundedpsbt

### PR DESCRIPTION
**Pull request by Sjors**
Backport of #15911 for 0.18 branch.

The original PR changed `rawtransaction_util.cpp`, whereas the backport changes `rawtransaction.cpp` to avoid having to also backport #15638.

### CODE REVIEW

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -528,7 +526,11 @@ static UniValue createrawtransaction(const JSONRPCRequest& request)
         }, true
     );
 
-    CMutableTransaction rawTx = ConstructTransaction(request.params[0], request.params[1], request.params[2], request.params[3]);
+    bool rbf = false;
+    if (!request.params[3].isNull()) {
+     rbf = request.params[3].isTrue();
```
Comment (meshcollider) [inline src/rpc/rawtransaction.cpp]: nit: indentation

**Review by meshcollider — COMMENTED** [review]
I have verified that the code changes here match those in the original PR (haven't actually re-reviewed the changes though, just verified the backport)

a678e06874e0004cd026e1969c27ecfc643ac289

File: src/rpc/rawtransaction.cpp (hunk 1 above)
Comment (Sjors) [reply src/rpc/rawtransaction.cpp]: Fix

**Comment by meshcollider** [comment]
re-utACK 576580fe8a063f21c0e903af1cf8f85cd6cb71d7

**Comment by MarcoFalke** [comment]
ACK 576580fe8a063f21c0e903af1cf8f85cd6cb71d7

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 576580fe8a063f21c0e903af1cf8f85cd6cb71d7
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUi90wwAwVZA8P1MHIw8biLLtDV+osnhJQ8yQCXAZG2ED4snoCbV5s2SczbTgksT
BWzEiCEQqf5s0lF1cgnCjkaYpIidpZ55O7iVSYTFnvlAd8Aw6cAH0Xe0V9VF+WhG
0tW3GBTIaOX4ZEfwMD0nSxdAorL1UWYt3HgDjaGNlZBWzScdFlTHzLb7LXKQqLL2
CprrQo/n2Ryp+riKB4998YserFpoudeHRcYzTR4aq0bmgfYEVtXZuxhlcz75BdQh
nw2Ehas82qqOtszuDUbvvOkZguwUfwEtoAxH3yUxsGUUwISOPmn2ekdE8u4WRxos
lgemrveGSKjPmEEhWKSETR97lTe6XLAjiaPtiClxAQSmBOhtuY0LqZpmhkh9B2Ru
R6QBnF+iZDAc7e0OQ7Bqug273Swhub5YbDS2qs2YL63EO+SKo528SNcznJRHBI8a
ZBdKWDxqMJysj5LkrW4oeIH312vcMK1wbBjiF5IZpmalShvHHux7oxPVFUpTEOfu
ywn7IU9C
=3lWE
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `f5a11e9d65e3f6cfd270f092131edbd8b4b33243c1a93c5ebd8c3244f41517df  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108f5a11e9d65e3f6cfd270f092131edbd8b4b33243c1a93c5ebd8c3244f41517dff0108bbcea5fe02a1bfdfcf3e10fcbf23cfa08fff01074ee649c00a2701d38c64fb327d2b87e08f1045d5a95abf008ad57843549e0ad010083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff01083fc33a271906fe8317935facfbbb1ff08f1045d5a95abf008c20cb4c5b7a6543d0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010a7c8da5e193d550a4feadaf04f4f2dec08f1045d5a95abf00811f98b347c6690860083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6df010e2684733d00f2d71323d0b758c6175bc08f1045d5a95abf008d4ba0b9b22f24d870083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Comment by fanquake** [comment]
Weren't all the commits here missing the backport metadata?

**Comment by Sjors** [comment]
Probably, what backport metadata should be in a commit?

**Comment by MarcoFalke** [comment]
Maybe we should mention the backport script in the contrib/devtools readme?

**Comment by MarcoFalke** [comment]
The code itself was the same when I did the backport, but those were the commit differences (via range-diff):


```
1:  ee950ec465 ! 1:  1057254270 [rpc] walletcreatefundedpsbt: use wallet default RBF
    @@ -2,6 +2,9 @@
     
         [rpc] walletcreatefundedpsbt: use wallet default RBF
     
    +    Github-Pull: #15911
    +    Rebased-From: 4fcb698bc2bb74171cd3a14b94f9882d8e19e9fb
    +
      diff --git a/src/rpc/rawtransaction.cpp b/src/rpc/rawtransaction.cpp
      --- a/src/rpc/rawtransaction.cpp
      +++ b/src/rpc/rawtransaction.cpp
    @@ -84,6 +87,15 @@
      --- a/src/wallet/rpcwallet.cpp
      +++ b/src/wallet/rpcwallet.cpp
     @@
    +                                     {"vout_index", RPCArg::Type::NUM, RPCArg::Optional::OMITTED, "The zero-based output index, before a change output is added."},
    +                                 },
    +                             },
    +-                            {"replaceable", RPCArg::Type::BOOL, /* default */ "false", "Marks this transaction as BIP125 replaceable.\n"
    ++                            {"replaceable", RPCArg::Type::BOOL, /* default */ "wallet default", "Marks this transaction as BIP125 replaceable.\n"
    +                             "                              Allows this transaction to be replaced by a transaction with higher fees"},
    +                             {"conf_target", RPCArg::Type::NUM, /* default */ "Fallback to wallet's confirmation target", "Confirmation target (in blocks)"},
    +                             {"estimate_mode", RPCArg::Type::STR, /* default */ "UNSET", "The fee estimate mode, must be one of:\n"
    +@@
      
          CAmount fee;
          int change_position;
    @@ -119,12 +131,12 @@
              block_height = self.nodes[0].getblockcount()
              unspent = self.nodes[0].listunspent()[0]
     -        psbtx_info = self.nodes[0].walletcreatefundedpsbt([{"txid":unspent["txid"], "vout":unspent["vout"]}], [{self.nodes[2].getnewaddress():unspent["amount"]+1}], block_height+2, {"replaceable":True}, False)
    -+        psbtx_info = self.nodes[0].walletcreatefundedpsbt([{"txid":unspent["txid"], "vout":unspent["vout"]}], [{self.nodes[2].getnewaddress():unspent["amount"]+1}], block_height+2, {"replaceable":False}, False)
    ++        psbtx_info = self.nodes[0].walletcreatefundedpsbt([{"txid":unspent["txid"], "vout":unspent["vout"]}], [{self.nodes[2].getnewaddress():unspent["amount"]+1}], block_height+2, {"replaceable": False}, False)
              decoded_psbt = self.nodes[0].decodepsbt(psbtx_info["psbt"])
              for tx_in, psbt_in in zip(decoded_psbt["tx"]["vin"], decoded_psbt["inputs"]):
     -           assert_equal(tx_in["sequence"], MAX_BIP125_RBF_SEQUENCE)
     -           assert "bip32_derivs" not in psbt_in
    -+            assert(tx_in["sequence"] > MAX_BIP125_RBF_SEQUENCE)
    ++            assert tx_in["sequence"] > MAX_BIP125_RBF_SEQUENCE
     +            assert "bip32_derivs" not in psbt_in
              assert_equal(decoded_psbt["tx"]["locktime"], block_height+2)
      
2:  0942a60c06 ! 2:  7ab99cf477 [doc] rpc: remove "fallback to" from RBF default help
    @@ -2,6 +2,9 @@
     
         [doc] rpc: remove "fallback to" from RBF default help
     
    +    Github-Pull: #15911
    +    Rebased-From: 9ed062b5685eb6227d694572cb0f7bfbcc151b36
    +
      diff --git a/src/wallet/rpcwallet.cpp b/src/wallet/rpcwallet.cpp
      --- a/src/wallet/rpcwallet.cpp
      +++ b/src/wallet/rpcwallet.cpp
    @@ -48,12 +51,3 @@
                                  {"totalFee", RPCArg::Type::NUM, /* default */ "fallback to 'confTarget'", "Total fee (NOT feerate) to pay, in satoshis.\n"
                  "                         In rare cases, the actual fee paid might be slightly higher than the specified\n"
                  "                         totalFee if the tx change output has to be removed because it is too close to\n"
    -@@
    -                                     {"vout_index", RPCArg::Type::NUM, RPCArg::Optional::OMITTED, "The zero-based output index, before a change output is added."},
    -                                 },
    -                             },
    --                            {"replaceable", RPCArg::Type::BOOL, /* default */ "false", "Marks this transaction as BIP125 replaceable.\n"
    -+                            {"replaceable", RPCArg::Type::BOOL, /* default */ "wallet default", "Marks this transaction as BIP125 replaceable.\n"
    -                             "                              Allows this transaction to be replaced by a transaction with higher fees"},
    -                             {"conf_target", RPCArg::Type::NUM, /* default */ "Fallback to wallet's confirmation target", "Confirmation target (in blocks)"},
    -                             {"estimate_mode", RPCArg::Type::STR, /* default */ "UNSET", "The fee estimate mode, must be one of:\n"
3:  576580fe8a ! 3:  86bc93ec44 [test] walletcreatefundedpsbt: check RBF is disabled when -walletrbf=0
    @@ -2,19 +2,13 @@
     
         [test] walletcreatefundedpsbt: check RBF is disabled when -walletrbf=0
     
    +    Github-Pull: #15911
    +    Rebased-From: d6b3640ac732f6f66a8cb6761084d1beecc8a876
    +
      diff --git a/test/functional/rpc_psbt.py b/test/functional/rpc_psbt.py
      --- a/test/functional/rpc_psbt.py
      +++ b/test/functional/rpc_psbt.py
     @@
    - 
    - from decimal import Decimal
    - from test_framework.test_framework import BitcoinTestFramework
    --from test_framework.util import assert_equal, assert_raises_rpc_error, connect_nodes_bi, disconnect_nodes, find_output, sync_blocks
    -+from test_framework.util import assert_equal, assert_greater_than, assert_raises_rpc_error, connect_nodes_bi, disconnect_nodes, find_output, sync_blocks
    - 
    - import json
    - import os
    -@@
              self.num_nodes = 3
              self.extra_args = [
                  ["-walletrbf=1"],
    @@ -24,10 +18,10 @@
              ]
      
     @@
    -         psbtx_info = self.nodes[0].walletcreatefundedpsbt([{"txid":unspent["txid"], "vout":unspent["vout"]}], [{self.nodes[2].getnewaddress():unspent["amount"]+1}], block_height+2, {"replaceable":False}, False)
    +         psbtx_info = self.nodes[0].walletcreatefundedpsbt([{"txid":unspent["txid"], "vout":unspent["vout"]}], [{self.nodes[2].getnewaddress():unspent["amount"]+1}], block_height+2, {"replaceable": False}, False)
              decoded_psbt = self.nodes[0].decodepsbt(psbtx_info["psbt"])
              for tx_in, psbt_in in zip(decoded_psbt["tx"]["vin"], decoded_psbt["inputs"]):
    --            assert(tx_in["sequence"] > MAX_BIP125_RBF_SEQUENCE)
    +-            assert tx_in["sequence"] > MAX_BIP125_RBF_SEQUENCE
     +            assert_greater_than(tx_in["sequence"], MAX_BIP125_RBF_SEQUENCE)
                  assert "bip32_derivs" not in psbt_in
              assert_equal(decoded_psbt["tx"]["locktime"], block_height+2)

**Comment by meshcollider** [comment]
Yeah the Github-Pull: and Rebased-From: were missing

**Comment by MarcoFalke** [comment]
See https://github.com/bitcoin-core/bitcoin-maintainer-tools#backport for the script to backport

**Pull request merged**